pkg/volume/volume.go

```diff
@@ -132,11 +132,39 @@ type Attacher interface {
 	Attach() error
 }
 
+type PhysicalAttacher interface {
+	// WaitForAttach examines the file system for the device; if
+	// found that means the disk is attached and this function
+	// returns otherwise it polls until the device appears. If the
+	// device is not attached within the timeout period an error
+	// is returned.
+	WaitForAttach() error
```

Comment:
Maybe pass the timeout period as a parameter so that it can be controlled in a common place rather than as a one-off for each plugin. Honestly, I could go either way on this: it could be argued that each plugin should define what this period is independently (since different types of volumes could take a different amount of for this operation to complete).